src/Invictus.Testing/LogicAppClient.cs

```diff
@@ -0,0 +1,498 @@
+                });
+        }
+
+        /// <summary>
+        /// Updates the current JSON logic app definition with the given <paramref name="logicAppDefinition"/>,
+        /// and removes this update after the returned instance gets disposed.
+        /// </summary>
+        /// <param name="logicAppDefinition">Then JSON representation of the new definition.</param>
+        /// <returns>
+        ///     An instance to control the removal of the updates.
+        /// </returns>
+        public async Task<IAsyncDisposable> TemporaryUpdateAsync(string logicAppDefinition)
+        {
+            Guard.NotNull(logicAppDefinition, nameof(logicAppDefinition));
+
+            Workflow workflow = await _logicManagementClient.Workflows.GetAsync(_resourceGroup, _logicAppName);
+            object originalAppDefinition = workflow.Definition;
+
+            _logger.LogTrace("Updates (+) the logic app '{LogicAppName}' workflow definition in resource group '{ResourceGroup}'", _logicAppName, _resourceGroup);
+            await UpdateAsync(workflow, JObject.Parse(logicAppDefinition));
+            return AsyncDisposable.Create(async () =>
+            {
+                _logger.LogTrace("Reverts (-) the update of the logic app '{LogicAppName}' workflow definition in resource group '{ResourceGroup}'", _logicAppName, _resourceGroup);
+                await UpdateAsync(workflow, originalAppDefinition);
+            });
+        }
+
+        private async Task UpdateAsync(Workflow workflow, object logicAppDefinition)
+        {
+            workflow.Definition = logicAppDefinition;
+            Workflow resultWorkflow = 
+                await _logicManagementClient.Workflows.CreateOrUpdateAsync(_resourceGroup, _logicAppName, workflow);
+
+            if (resultWorkflow.Name != _logicAppName)
+            {
+                throw new InvalidOperationException(
+                    $"Could not update the logic app '{_logicAppName}' workflow in resource group '{_resourceGroup}'correctly, "
+                    + "please make sure the authentication on this resource is set correctly and has the right access to this resource");
+            }
+        }
+
+        /// <summary>
+        /// Deletes the current logic app resource on Azure.
+        /// </summary>
+        public async Task DeleteAsync()
+        {
+            _logger.LogTrace("Deletes the workflow of logic app '{LogicAppName}' in resource group '{ResourceGroup}'", _logicAppName, _resourceGroup);
+            await _logicManagementClient.Workflows.DeleteAsync(_resourceGroup, _logicAppName);
+        }
+
+        /// <summary>
+        /// Runs the current logic app resource by searching for triggers on the logic app.
+        /// </summary>
+        /// <exception cref="LogicAppTriggerNotFoundException">When no trigger can be found on the logic app.</exception>
+        public async Task RunAsync()
```

Comment:
Can you provide a test that uses this overload please?

Reply:
Hmm, what does this do actually? I mean 'running the logic app'? Is there something I can use to check if it's running or not? And additionally, should we have then a 'stop' too?

Reply:
It starts a Logic App without you having to worry about the trigger name; that's what I think based on the implementation.

Is that correct @mbraekman ?

Reply:
What the RunAsync can do is trigger a Logic App to start running without specifying the name of the trigger.
In most cases a Logic App will only have a single trigger, so it would be unnecessary to have the developer specify its name. However, the LogicManagementClient-class does require us to provide the name of the trigger. Hence the test-framework doing the lookup for us.

Can you check if it is running?
I should check in the actual code or docs of the LogicManagementClient, but if I remember this correctly, it should return a correlationId, which in turn can be used to retrieve a Logic App run.

Can you stop the run?
Once you're able to capture the LogicAppRun-object, it supports a Cancel(Async)-method, you can use to cancel the run.